### PR TITLE
5g review hep

### PR DESCRIPTION
**Pull request by heplesser**
- grow() now with max-blocksize strategy
- reserve_connection() does nothing
- various other changes

### CODE REVIEW

File: libnestutil/vector_util.h [hunk 1]
```diff
@@ -32,11 +32,14 @@ template < typename T >
 inline void
 grow( std::vector< T >& v )
 {
-  // use 1.5 growth strategy (see, e.g.,
-  // https://github.com/facebook/folly/blob/master/folly/docs/FBVector.md)
+  const size_t max_block_size_MiB = 256;
```
Comment (jakobj) [inline libnestutil/vector_util.h]: maybe add a comment why we chose 256?

File: nestkernel/connection_manager.cpp [hunk 2]
```diff
@@ -1488,11 +1488,23 @@ nest::ConnectionManager::reserve_connections( const thread tid,
   const synindex syn_id,
   const size_t count )
 {
+  /*
```
Comment (jakobj) [inline nestkernel/connection_manager.cpp]: do i understand correctly that we want to keep the code above that computes the expected number of connections, and also this todo in the release?

**Comment by jakobj** [comment]
@hakonsbm @stinebuu @suku248 since @heplesser is on holiday, maybe one of you can address my comments so we can move on?

**Comment by stinebuu** [comment]
@jakobj I unfortunately don't remember exactly why 256 was chosen, so I don't know if I am much help there.. I am guessing the TODO and code should stay put. Sorry for not being more helpful!

**Comment by jakobj** [comment]
@stinebuu no problem, thanks! @suku248 do you remember why this number was chosen? I imagine this was by educated guessing + trial-and-error?

**Comment by suku248** [comment]
Yes, the tests showed less memory usage for factor 2 growth and max block size of 256 than for factor 1.5 growth and max block size of 128.

**Comment by jakobj** [comment]
OK, thanks. I'll add that information. Travis seems to have failed just because of a travis error, not because of nest.

**Pull request merged**